### ISSUE TITLE
Not the same number of samples at each epoch

### ISSUE THREAD

**Issue by loiccordone** [MaxHalford/pytorch-resample#2]
Hello,

I have classification dataset of 32k samples, where 89% of my data is class 0 and the other 11% is class 1. I tried to use pytorch-resample to Undersample my 0-class, with a distribution of 0.5/0.5. It did work great, but I noticed that the dataset differs at each epoch, making the training useless. I know there is a random seed initialized the first time we call the Undersample class, but I didn't manage to reset this seed after each epoch since the dataset is part of the DataLoader. 

Have you not encountered a similar behavior? Do you have an idea for a fix? Thanks in advance

**Comment by MaxHalford** [comment]
Hello! Glad to know you're using the package.

You could reset the seed by replacing the `rng` parameter before generating a new batch, but that would be hacky from an API perspective.

Could you maybe provide a minimal reproducible example to further explain what you would like? I'm happy to make changes that help you but I'm not 100% clear about what you want.

**Comment by loiccordone** [comment]
Hello, thanks for releasing it!

After several tries, I can't seem to reproduce the problem with random data, so it seems to come from the [WebDataset](https://github.com/tmbdev/webdataset) package. I've tried to reseed the rng, and aside from being hacky it still doesn't work with my WebDataset.
